imod_coupler/drivers/ribametamod/ribametamod.py

```diff
@@ -285,8 +286,33 @@ def couple(self) -> None:
             ),
         )
 
-        # Set exchange-class to handle all exchanges to Ribasim Basins
         if self.has_ribasim:
+            if self.has_metaswap:
+                if self.coupling.rib_msw_sprinkling_map_surface_water is not None:
+                    self.enable_sprinkling_surface_water = True
+                    self.ribasim_user_demand = self.ribasim.get_value_ptr(
+                        "user_demand.demand"
+                    )
+                    modribmsw_arrays["rib_sprinkling_demand"] = self.ribasim_user_demand
+                    n_users = np.shape(self.mapping.msw2rib["sw_sprinkling"])[0]
+                    n_priorities = np.size(self.ribasim_user_demand) // n_users
```

Comment:
For now, i think we best raise an exception if multiple priorities per MetaSwap user are defined.